mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java

```diff
@@ -1593,4 +1595,35 @@ public Source resolve(String href, String base) throws TransformerException {
         }
 
     }
+    
+    private static class MCRI18NResolver implements URIResolver {
+        /**
+         * Resolves the I18N String value for the given property.<br><br>
+         * 
+         * Syntax: <code>i18n:{i18n-property}</code>
+         * 
+         * @param href
+         *            URI in the syntax above
+         * @param base
+         *            not used
+         * 
+         * @return the root element of the XML document
+         * @see javax.xml.transform.URIResolver
+         */
+        @Override
+        public Source resolve(String href, String base) throws TransformerException {
+            LOGGER.debug("start resolving " + href);
+            StringTokenizer pst = new StringTokenizer(href, ":", true);
+            if (pst.countTokens() != 3) {
+                throw new IllegalArgumentException("Invalid format of uri for retrieval of i18n: " + href);
+            }
+            pst.nextToken(); // "i18n"
+            pst.nextToken(); // :
+            String property = pst.nextToken();
+            Element root = new Element("i18n");
+            root.setText(MCRTranslation.translate(property, MCRTranslation.getCurrentLocale()));
```

Comment:
Would an additional feature make sense, that allows to specify another language in the URI?